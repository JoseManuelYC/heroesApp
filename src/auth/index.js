export * from "./pages/Login";
export * from "./context";
export * from "./types/types";
