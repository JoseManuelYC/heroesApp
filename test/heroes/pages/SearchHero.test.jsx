
import { fireEvent, render, screen } from "@testing-library/react";
import  { SearchHero }  from "../../../src/heroes";
import { MemoryRouter } from "react-router-dom";

const mockSearchValue = jest.fn();

jest.mock('react-router-dom', ()=>({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockSearchValue,
}))

describe('Pruebas en el FC SearchHero', () => {
  beforeEach(() => jest.clearAllMocks());
  test('Debe de mostrarse correctamente con valores por defecto ', () => {
    const { container } = 
    render(
        <MemoryRouter>
          <SearchHero />
        </MemoryRouter>
      )
    expect(container).toMatchSnapshot();
  })
  test('Debe de mostrar el valor del query y el input el value con la query', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']} >
        <SearchHero />
      </MemoryRouter>
    )
    const input = screen.getByRole("textbox");
    expect(input.value).toBe("batman");
    const img = screen.getByRole("img");
    expect(img.src).toContain("/heroes/dc-batman.jpg");
    const alert = screen.getByLabelText("danger");
    expect(alert.style.display).toBe("none");
  })
  test('Debe de mostrar error si no se muestra el Hero', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123']} >
        <SearchHero />
      </MemoryRouter>
    )
    const alert = screen.getByLabelText("danger");
    expect(alert.style.display).toBe("");
  })
  test('Debe de llamar el navigate a la pantalla nueva', () => {
    render(
      <MemoryRouter initialEntries={['/search']} >
        <SearchHero />
      </MemoryRouter>
    )
    const formelement = screen.getByRole("textbox");
    fireEvent.change( formelement, {target : {name:"searchText", value:"superman"}} );
    const form = screen.getByLabelText("form");
    fireEvent.submit(form);
    expect(mockSearchValue).toHaveBeenCalledWith("?q=superman");
  })
  
  
})
