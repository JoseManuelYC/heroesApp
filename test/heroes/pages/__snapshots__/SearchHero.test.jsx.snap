// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el FC SearchHero Debe de mostrarse correctamente con valores por defecto  1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-5 mt-2"
    >
      <h3>
        Search
      </h3>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-3"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7 mt-2"
    >
      <h3>
        Results
      </h3>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
        style="display: none;"
      >
        Search a hero
      </div>
      <div
        aria-label="danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        The Hero: 
        <b />
         not found
      </div>
    </div>
  </div>
</div>
`;
